src/pages/SignUpResearcher.js

```diff
@@ -44,13 +49,21 @@ function SignUpResearcher() {
     return errors;
   };
 
-  const handleSubmit = (values) => {
-    //TODO: authenticate to allow login
-
-    // Db.addResearchData(useContext(UserContext)?.uid, ...); TODO
-    // navigate to next page
-
+  const handleSubmit = async values => {
+    try {
+      await Db.addResearchData(userId, values);
```

Comment:
The await does nothing here actually because the function is not returning a Promise , will explain more in my fix!